test/unit/logger-client.js

```diff
@@ -0,0 +1,64 @@
+'use strict'
+
+// Testing env vars is painful here. Override the default and check config-driven
+// option later
+process.env.LOGGER_SEND_USER_AGENT = ''
+
+const {test} = require('tap')
+const loggerClient = require('../../lib/logger-client.js')
+const config = require('../../lib/config.js')
+
+const CONFIG_DEFAULT = {
+  ...config
+, key: 'abc123'
+, UserAgent: 'fake-transport@1.2.3'
+, hostname: 'myhostname'
+}
+
+test('Check logger defaults', async (t) => {
+  const logger = loggerClient.createLoggerClient(CONFIG_DEFAULT)
+  const requestDefaults = logger[Symbol.for('requestDefaults')]
+
+  t.match(requestDefaults.headers, {
+    'Content-Type': 'application/json; charset=UTF-8'
+  , 'Content-Encoding': 'gzip'
+  }, 'Default headers are correct')
+
+  const expectedAgentHeader = /@logdna\/logger\/\d+\.\d+\.\d+ (fake-transport@1\.2\.3)/
+  t.match(
+    logger
+  , {
+      [Symbol.for('userAgentHeader')]: expectedAgentHeader
+    }
+  , 'UserAgent header is correct'
+  )
```

Comment:
I don't think `match` will work with symbols since they are not enumerable
```js
test('foo()', async (t) => {
  const k = Symbol('foo')
  const o = {[k]: true}
  t.match(o, {[k]: false}) // passes
})
```

Reply:
Hrm, well doing a `match` on the string value gave me plenty of problems.  Not sure what to do here.

Reply:
fixed with a normal regex `.test`